src/platform/activity/activity.service.ts

```diff
@@ -80,6 +84,40 @@ export class ActivityService {
     });
   }
 
+  public async getPaginatedActivity(
+    collaborationIDs: string[],
+    options?: {
+      types?: ActivityEventType[];
+      visibility?: boolean;
+      userID?: string;
+      paginationArgs?: PaginationArgs;
+    }
+  ) {
+    const {
+      types,
+      visibility = true,
+      userID,
+      paginationArgs = {},
+    } = options ?? {};
+
+    const qb = await this.activityRepository.createQueryBuilder('activity');
+
+    qb.where({
+      collaborationID: In(collaborationIDs),
+      visibility: visibility,
+    }).orderBy({ createdDate: 'DESC' });
```

Comment:
I would add an enum (or constants?) `ASC` and `DESC` that are re-shared in the application and would provide them as arguments. It may be that we need things ordered differently in the future.

Reply:
As a rule of thumb I use is to not over engineer things on the first try.
It's already using the type `OrderByCondition`, or a simple `'ASC' | 'DESC'` will be enough.

I have added it as an option.